requirements.txt

```diff
@@ -0,0 +1,6 @@
+fastapi~=0.111.1
```

Comment:
Here is the complete list of dependencies, I don't understand why there are so many, I created the project with 
``
FROM continuumio/miniconda3
RUN conda install conda-forge::rdkit
``
In the last homework, to include all of these in requirements.txt, I ran ``pip freeze > requirements.txt``, but did not do it this time. Any ideas?

#### Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
annotated-types           0.7.0                    pypi_0    pypi
anyio                     4.4.0                    pypi_0    pypi
brotli                    1.1.0                hd590300_1    conda-forge
brotli-bin                1.1.0                hd590300_1    conda-forge
bzip2                     1.0.8                h4bc722e_7    conda-forge
ca-certificates           2024.7.4             hbcca054_0    conda-forge
cairo                     1.18.0               hebfffa5_3    conda-forge
certifi                   2024.7.4           pyhd8ed1ab_0    conda-forge
chardet                   5.2.0           py312h7900ff3_1    conda-forge
click                     8.1.7                    pypi_0    pypi
contourpy                 1.2.1           py312h8572e83_0    conda-forge
cycler                    0.12.1             pyhd8ed1ab_0    conda-forge
dnspython                 2.6.1                    pypi_0    pypi
email-validator           2.2.0                    pypi_0    pypi
expat                     2.6.2                h59595ed_0    conda-forge
fastapi                   0.111.1                  pypi_0    pypi
fastapi-cli               0.0.4                    pypi_0    pypi
font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
font-ttf-inconsolata      3.000                h77eed37_0    conda-forge
font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge
font-ttf-ubuntu           0.83                 h77eed37_2    conda-forge
fontconfig                2.14.2               h14ed4e7_0    conda-forge
fonts-conda-ecosystem     1                             0    conda-forge
fonts-conda-forge         1                             0    conda-forge
fonttools                 4.53.1          py312h41a817b_0    conda-forge
freetype                  2.12.1               h267a509_2    conda-forge
freetype-py               2.3.0              pyhd8ed1ab_0    conda-forge
greenlet                  3.0.3           py312h30efb56_0    conda-forge
h11                       0.14.0                   pypi_0    pypi
httpcore                  1.0.5                    pypi_0    pypi
httptools                 0.6.1                    pypi_0    pypi
httpx                     0.27.0                   pypi_0    pypi
icu                       75.1                 he02047a_0    conda-forge
idna                      3.7                      pypi_0    pypi
iniconfig                 2.0.0                    pypi_0    pypi
jinja2                    3.1.4                    pypi_0    pypi
kiwisolver                1.4.5           py312h8572e83_1    conda-forge
lcms2                     2.16                 hb7c19ff_0    conda-forge
ld_impl_linux-64          2.40                 hf3520f5_7    conda-forge
lerc                      4.0.0                h27087fc_0    conda-forge
libblas                   3.9.0           22_linux64_openblas    conda-forge
libboost                  1.84.0               h0ccab89_4    conda-forge
libboost-python           1.84.0          py312hf74af30_4    conda-forge
libbrotlicommon           1.1.0                hd590300_1    conda-forge
libbrotlidec              1.1.0                hd590300_1    conda-forge
libbrotlienc              1.1.0                hd590300_1    conda-forge
libcblas                  3.9.0           22_linux64_openblas    conda-forge
libdeflate                1.20                 hd590300_0    conda-forge
libexpat                  2.6.2                h59595ed_0    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 14.1.0               h77fa898_0    conda-forge
libgfortran-ng            14.1.0               h69a702a_0    conda-forge
libgfortran5              14.1.0               hc5f4f2c_0    conda-forge
libglib                   2.80.3               h8a4344b_1    conda-forge
libgomp                   14.1.0               h77fa898_0    conda-forge
libiconv                  1.17                 hd590300_2    conda-forge
libjpeg-turbo             3.0.0                hd590300_1    conda-forge
liblapack                 3.9.0           22_linux64_openblas    conda-forge
libnsl                    2.0.1                hd590300_0    conda-forge
libopenblas               0.3.27          pthreads_hac2b453_1    conda-forge
libpng                    1.6.43               h2797004_0    conda-forge
libsqlite                 3.46.0               hde9e2c9_0    conda-forge
libstdcxx-ng              14.1.0               hc0a3c3a_0    conda-forge
libtiff                   4.6.0                h1dd3fc0_3    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libwebp-base              1.4.0                hd590300_0    conda-forge
libxcb                    1.16                 hd590300_0    conda-forge
libxcrypt                 4.4.36               hd590300_1    conda-forge
libzlib                   1.3.1                h4ab18f5_1    conda-forge
markdown-it-py            3.0.0                    pypi_0    pypi
markupsafe                2.1.5                    pypi_0    pypi
matplotlib-base           3.9.1           py312h9201f00_0    conda-forge
mdurl                     0.1.2                    pypi_0    pypi
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
ncurses                   6.5                  h59595ed_0    conda-forge
numpy                     2.0.0           py312h22e1c76_0    conda-forge
openjpeg                  2.5.2                h488ebb8_0    conda-forge
openssl                   3.3.1                h4bc722e_2    conda-forge
packaging                 24.1               pyhd8ed1ab_0    conda-forge
pandas                    2.2.2           py312h1d6d2e6_1    conda-forge
pcre2                     10.44                h0f59acf_0    conda-forge
pillow                    10.4.0          py312h287a98d_0    conda-forge
pip                       24.0               pyhd8ed1ab_0    conda-forge
pixman                    0.43.2               h59595ed_0    conda-forge
pluggy                    1.5.0                    pypi_0    pypi
pthread-stubs             0.4               h36c2ea0_1001    conda-forge
pycairo                   1.26.1          py312h3bc4990_0    conda-forge
pydantic                  2.8.2                    pypi_0    pypi
pydantic-core             2.20.1                   pypi_0    pypi
pygments                  2.18.0                   pypi_0    pypi
pyparsing                 3.1.2              pyhd8ed1ab_0    conda-forge
pytest                    8.3.1                    pypi_0    pypi
python                    3.12.4          h194c7f8_0_cpython    conda-forge
python-dateutil           2.9.0              pyhd8ed1ab_0    conda-forge
python-dotenv             1.0.1                    pypi_0    pypi
python-multipart          0.0.9                    pypi_0    pypi
python-tzdata             2024.1             pyhd8ed1ab_0    conda-forge
python_abi                3.12                    4_cp312    conda-forge
pytz                      2024.1             pyhd8ed1ab_0    conda-forge
pyyaml                    6.0.1                    pypi_0    pypi
qhull                     2020.2               h434a139_5    conda-forge
rdkit                     2024.03.4       py312hf6fb7c8_1    conda-forge
readline                  8.2                  h8228510_1    conda-forge
reportlab                 4.2.2           py312h9a8786e_0    conda-forge
rich                      13.7.1                   pypi_0    pypi
rlpycairo                 0.2.0              pyhd8ed1ab_0    conda-forge
setuptools                71.0.4             pyhd8ed1ab_0    conda-forge
shellingham               1.5.4                    pypi_0    pypi
six                       1.16.0             pyh6c4a22f_0    conda-forge
sniffio                   1.3.1                    pypi_0    pypi
sqlalchemy                2.0.31          py312h9a8786e_0    conda-forge
starlette                 0.37.2                   pypi_0    pypi
tk                        8.6.13          noxft_h4845f30_101    conda-forge
typer                     0.12.3                   pypi_0    pypi
typing-extensions         4.12.2               hd8ed1ab_0    conda-forge
typing_extensions         4.12.2             pyha770c72_0    conda-forge
tzdata                    2024a                h0c530f3_0    conda-forge
uvicorn                   0.30.3                   pypi_0    pypi
uvloop                    0.19.0                   pypi_0    pypi
watchfiles                0.22.0                   pypi_0    pypi
websockets                12.0                     pypi_0    pypi
wheel                     0.43.0             pyhd8ed1ab_1    conda-forge
xorg-kbproto              1.0.7             h7f98852_1002    conda-forge
xorg-libice               1.1.1                hd590300_0    conda-forge
xorg-libsm                1.2.4                h7391055_0    conda-forge
xorg-libx11               1.8.9                hb711507_1    conda-forge
xorg-libxau               1.0.11               hd590300_0    conda-forge
xorg-libxdmcp             1.1.3                h7f98852_0    conda-forge
xorg-libxext              1.3.4                h0b41bf4_2    conda-forge
xorg-libxrender           0.9.11               hd590300_0    conda-forge
xorg-renderproto          0.11.1            h7f98852_1002    conda-forge
xorg-xextproto            7.3.0             h0b41bf4_1003    conda-forge
xorg-xproto               7.0.31            h7f98852_1007    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
zlib                      1.3.1                h4ab18f5_1    conda-forge
zstd                      1.5.6                ha6fb4c9_0    conda-forge

Reply:
This is the result of `conda list --export`
```
#This file may be used to create an environment using:
# conda create --name <env> --file <this file>
# platform: linux-64
_libgcc_mutex=0.1=conda_forge
_openmp_mutex=4.5=2_gnu
annotated-types=0.7.0=pyhd8ed1ab_0
anyio=4.4.0=pyhd8ed1ab_0
brotli=1.1.0=hd590300_1
brotli-bin=1.1.0=hd590300_1
bzip2=1.0.8=h4bc722e_7
ca-certificates=2024.7.4=hbcca054_0
cairo=1.18.0=hebfffa5_3
certifi=2024.7.4=pyhd8ed1ab_0
chardet=5.2.0=py312h7900ff3_1
click=8.1.7=unix_pyh707e725_0
contourpy=1.2.1=py312h8572e83_0
cycler=0.12.1=pyhd8ed1ab_0
dnspython=2.6.1=pyhd8ed1ab_1
email-validator=2.2.0=pyhd8ed1ab_0
email_validator=2.2.0=hd8ed1ab_0
exceptiongroup=1.2.2=pyhd8ed1ab_0
expat=2.6.2=h59595ed_0
fastapi=0.111.1=pyhd8ed1ab_0
fastapi-cli=0.0.4=pyhd8ed1ab_0
font-ttf-dejavu-sans-mono=2.37=hab24e00_0
font-ttf-inconsolata=3.000=h77eed37_0
font-ttf-source-code-pro=2.038=h77eed37_0
font-ttf-ubuntu=0.83=h77eed37_2
fontconfig=2.14.2=h14ed4e7_0
fonts-conda-ecosystem=1=0
fonts-conda-forge=1=0
fonttools=4.53.1=py312h41a817b_0
freetype=2.12.1=h267a509_2
freetype-py=2.3.0=pyhd8ed1ab_0
greenlet=3.0.3=py312h30efb56_0
h11=0.14.0=pyhd8ed1ab_0
h2=4.1.0=pyhd8ed1ab_0
hpack=4.0.0=pyh9f0ad1d_0
httpcore=1.0.5=pyhd8ed1ab_0
httpx=0.27.0=pyhd8ed1ab_0
hyperframe=6.0.1=pyhd8ed1ab_0
icu=75.1=he02047a_0
idna=3.7=pyhd8ed1ab_0
jinja2=3.1.4=pyhd8ed1ab_0
keyutils=1.6.1=h166bdaf_0
kiwisolver=1.4.5=py312h8572e83_1
krb5=1.21.3=h659f571_0
lcms2=2.16=hb7c19ff_0
ld_impl_linux-64=2.40=hf3520f5_7
lerc=4.0.0=h27087fc_0
libblas=3.9.0=23_linux64_openblas
libboost=1.84.0=h0ccab89_4
libboost-python=1.84.0=py312hf74af30_4
libbrotlicommon=1.1.0=hd590300_1
libbrotlidec=1.1.0=hd590300_1
libbrotlienc=1.1.0=hd590300_1
libcblas=3.9.0=23_linux64_openblas
libdeflate=1.20=hd590300_0
libedit=3.1.20191231=he28a2e2_2
libexpat=2.6.2=h59595ed_0
libffi=3.4.2=h7f98852_5
libgcc-ng=14.1.0=h77fa898_0
libgfortran-ng=14.1.0=h69a702a_0
libgfortran5=14.1.0=hc5f4f2c_0
libglib=2.80.3=h8a4344b_1
libgomp=14.1.0=h77fa898_0
libiconv=1.17=hd590300_2
libjpeg-turbo=3.0.0=hd590300_1
liblapack=3.9.0=23_linux64_openblas
libnsl=2.0.1=hd590300_0
libopenblas=0.3.27=pthreads_hac2b453_1
libpng=1.6.43=h2797004_0
libpq=16.3=ha72fbe1_0
librdkit=2024.03.5=h79cfef2_2
libsqlite=3.46.0=hde9e2c9_0
libstdcxx-ng=14.1.0=hc0a3c3a_0
libtiff=4.6.0=h1dd3fc0_3
libuuid=2.38.1=h0b41bf4_0
libwebp-base=1.4.0=hd590300_0
libxcb=1.16=hd590300_0
libxcrypt=4.4.36=hd590300_1
libzlib=1.3.1=h4ab18f5_1
markdown-it-py=3.0.0=pyhd8ed1ab_0
markupsafe=2.1.5=py312h98912ed_0
matplotlib-base=3.9.1=py312h9201f00_0
mdurl=0.1.2=pyhd8ed1ab_0
munkres=1.1.4=pyh9f0ad1d_0
ncurses=6.5=h59595ed_0
numpy=2.0.1=py312h1103770_0
openjpeg=2.5.2=h488ebb8_0
openssl=3.3.1=h4bc722e_2
packaging=24.1=pyhd8ed1ab_0
pandas=2.2.2=py312h1d6d2e6_1
pcre2=10.44=h0f59acf_0
pillow=10.4.0=py312h287a98d_0
pip=24.0=pyhd8ed1ab_0
pixman=0.43.2=h59595ed_0
pthread-stubs=0.4=h36c2ea0_1001
pycairo=1.26.1=py312h3bc4990_0
pydantic=2.8.2=pyhd8ed1ab_0
pydantic-core=2.20.1=py312hf008fa9_0
pygments=2.18.0=pyhd8ed1ab_0
pyparsing=3.1.2=pyhd8ed1ab_0
python=3.12.4=h194c7f8_0_cpython
python-dateutil=2.9.0=pyhd8ed1ab_0
python-multipart=0.0.9=pyhd8ed1ab_0
python-tzdata=2024.1=pyhd8ed1ab_0
python_abi=3.12=4_cp312
pytz=2024.1=pyhd8ed1ab_0
qhull=2020.2=h434a139_5
rdkit=2024.03.5=py312h7b4b7d0_2
readline=8.2=h8228510_1
reportlab=4.2.2=py312h9a8786e_0
rich=13.7.1=pyhd8ed1ab_0
rlpycairo=0.2.0=pyhd8ed1ab_0
setuptools=71.0.4=pyhd8ed1ab_0
shellingham=1.5.4=pyhd8ed1ab_0
six=1.16.0=pyh6c4a22f_0
sniffio=1.3.1=pyhd8ed1ab_0
sqlalchemy=2.0.31=py312h9a8786e_0
starlette=0.37.2=pyhd8ed1ab_0
tk=8.6.13=noxft_h4845f30_101
typer=0.12.3=pyhd8ed1ab_0
typer-slim=0.12.3=pyhd8ed1ab_0
typer-slim-standard=0.12.3=hd8ed1ab_0
typing-extensions=4.12.2=hd8ed1ab_0
typing_extensions=4.12.2=pyha770c72_0
tzdata=2024a=h0c530f3_0
uvicorn=0.30.3=py312h7900ff3_0
wheel=0.43.0=pyhd8ed1ab_1
xorg-kbproto=1.0.7=h7f98852_1002
xorg-libice=1.1.1=hd590300_0
xorg-libsm=1.2.4=h7391055_0
xorg-libx11=1.8.9=hb711507_1
xorg-libxau=1.0.11=hd590300_0
xorg-libxdmcp=1.1.3=h7f98852_0
xorg-libxext=1.3.4=h0b41bf4_2
xorg-libxrender=0.9.11=hd590300_0
xorg-renderproto=0.11.1=h7f98852_1002
xorg-xextproto=7.3.0=h0b41bf4_1003
xorg-xproto=7.0.31=h7f98852_1007
xz=5.2.6=h166bdaf_0
zlib=1.3.1=h4ab18f5_1
zstd=1.5.6=ha6fb4c9_0
```